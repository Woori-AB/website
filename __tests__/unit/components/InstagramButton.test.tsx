import { cleanup, render } from "@testing-library/react";
import { describe, it, vi, expect, afterEach } from "vitest";

describe("InstagramButton", () => {
  afterEach(() => {
    cleanup();
    vi.clearAllMocks();
  });

  it("matches snapshot(s)", async () => {
    // Arrange
    const { default: InstagramButton } = await vi.importActual<
      // eslint-disable-next-line @typescript-eslint/consistent-type-imports
      typeof import("@/components/InstagramButton")
    >("@/components/InstagramButton");

    // Act
    const { container } = render(<InstagramButton />);

    // Assert
    expect(container).toMatchSnapshot();
  });
});
