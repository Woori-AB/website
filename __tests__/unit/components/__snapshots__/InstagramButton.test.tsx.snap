// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InstagramButton > matches snapshot(s) 1`] = `
<div>
  <div
    class="bg-black/40 hover:bg-black/70 text-white rounded-4xl px-4 py-2"
  >
    <a
      class="flex gap-2"
      href="https://www.instagram.com/wooristockholm"
      rel="noopener noreferrer"
      target="_blank"
    >
      <mocked-lucide-react-instagram />
       
      mocked-i18n-'follow-on-instagram'
    </a>
  </div>
</div>
`;
